envoy/api/v2/auth/rbac.proto

```diff
@@ -0,0 +1,181 @@
+  // or "*.mtv.svc.cluster.local" (suffix match).
+  // If set to ["*"], it refers to all services in the namespace.
+  repeated string services = 1;
+
+  // Optional. A list of HTTP paths or gRPC methods.
+  // gRPC methods must be presented as fully-qualified name in the form of
+  // packageName.serviceName/methodName.
+  // Exact match, prefix match, and suffix match are supported for paths.
+  // For example, the path "/books/review" matches
+  // "/books/review" (exact match), or "/books/*" (prefix match),
+  // or "*/review" (suffix match).
+  // If not specified, it applies to any path.
+  repeated string paths = 2;
+
+  // Optional. A list of HTTP methods (e.g., "GET", "POST").
+  // It is ignored in gRPC case because the value is always "POST".
+  // If set to ["*"] or not specified, it applies to any method.
+  repeated string methods = 3;
+
+  // Definition of a custom constraint. The key in the constraint refers to an Istio attribute name
+  // (https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html).For example,
+  // destination.labels["app"], request.headers["version"].
+  message Constraint {
+    // Key of the constraint.
+    string key = 1;
+
+    // List of valid values for the constraint.
+    // Exact match, prefix match, and suffix match are supported for constraint values.
+    // For example, the value "v1alpha2" matches
+    // "v1alpha2" (exact match), or "v1*" (prefix match),
+    // or "*alpha2" (suffix match).
+    repeated string values = 2;
+  }
+
+  // Optional. Extra constraints in the ServiceRole specification.
+  // The above ServiceRole examples shows an example of constraint "version".
+  repeated Constraint constraints = 4;
+}
+
+// ServiceRoleBinding assigns a ServiceRole to a list of subjects.
+// This represents the "Spec" part of the ServiceRoleBinding object. The name and namespace
+// of the ServiceRoleBinding is specified in "metadata" section of the ServiceRoleBinding
+// object.
+message ServiceRoleBinding {
+  // Required. List of subjects that are assigned the ServiceRole object.
+  repeated Subject subjects = 1;
+
+  // Required. Reference to the ServiceRole object.
+  RoleRef role_ref = 2 [(validate.rules).message.required = true];
+}
+
+// Subject defines an identity or a group of identities. The identity is either a user or
+// a group or identified by a set of "properties". The names of a property
+// refers to an Istio attribute
+// (https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html). For example,
+// source.service, source.labels["app"], request.headers["version"].
+message Subject {
+  // Optional. The user name/ID that the subject represents.
+  string user = 1;
+
+  // Optional. The group that the subject belongs to.
+  string group = 2;
+
+  // Optional. The set of properties that identify the subject.
+  map<string, string> properties = 3;
+}
+
+// RoleRef refers to a role object.
+message RoleRef {
```

Comment:
I think I'm probably missing something, but how will this information actually be used by the inline filter? Logging? Settings role headers? If so can we document more clearly?